shell/src/main/java/alluxio/cli/bundler/command/CollectEnvCommand.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.bundler.command;
+
+import alluxio.client.file.FileSystemContext;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.AlluxioException;
+import alluxio.shell.CommandReturn;
+import alluxio.shell.ShellCommand;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Command to run a set of shell commands to get system information.
+ * */
+public class CollectEnvCommand extends AbstractInfoCollectorCommand {
```

Comment:
This command share a lot of common logic with `CollectAlluxioInfoCommand`. Can we move the shared logic to a base class and just define different shell commands in each sub class?

Reply:
`CollectAlluxioInfoCommand` now extends `CollectEnvCommand`. PTAL at the new logic. Thanks!

Reply:
It would be more maintainable to create a common base class for the two commands, e.g. `ExecuteShellScriptCollectorCommand`. Having `CollectAlluxioInfoCommand` extend `CollectEnvCommand` makes it harder to change `CollectEnvCommand` without breaking `CollectAlluxioInfoCommand` .